src/common/utils/get-differences/get.diff.ts

```
@@ -0,0 +1,39 @@
+ * @param {T} obj1 - The first object to compare.
+ * @param {T} obj2 - The second object to compare.
+ * @returns {Partial<T> | null} - A partial object containing the differences, or null if no differences are found.
+ */
+export const getDiff = <T extends Record<string, any>>(
+  obj1: T,
+  obj2: T
+): Partial<T> | null => {
+  const result: Partial<T> = {};
+
+  for (const key in obj1) {
+    if (obj1.hasOwnProperty(key) && obj2.hasOwnProperty(key)) {
```

Biome

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)